protocol-units/bridge/chains/ethereum/src/client.rs

```diff
@@ -342,7 +395,7 @@ impl BridgeContractCounterparty for EthClient {
 			FixedBytes(hash_lock.0),
 			U256::from(time_lock.0),
 			recipient.0 .0,
-			U256::from(amount.0),
+			U256::try_from(amount.0).unwrap(),
```

Comment:
We should not use `unwrap()` in production code. Instead you want to return a `BridgeContractInitiatorResult`

```
U256::try_from(amount.0).map_err(|e| {
        // Handle the error appropriately, possibly converting it to your custom error type
        BridgeContractInitiatorError::ConversionError
    })
```

And you would need to add `ConversionError` variant to `BridgeContractInitiatorError` Because:
`pub type BridgeContractInitiatorResult<T> = Result<T, BridgeContractInitiatorError>;`

https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err